gridpapers.dtx

```diff
@@ -505,8 +505,7 @@
   /pgf/pattern keys/myshift/.store in=\myshift,
   /pgf/pattern keys/myshift/.initial={(0,0)},
 }
-\tikzdeclarepattern{
-  name=lightcones,
+\tikzdeclarepattern{name=lightcones,
```

Comment:
This linebreak removal was not needed

Reply:
I just looked at your issue #21 and saw your comment about whitespace sensitivity. Is this what you were referring to?

Reply:
Yes; \tikzdeclarepattern can't handle whitespace before name= on old versions. I get "pattern lightcones not defined" (paraphrased) if I include the linebreak.

Reply:
That is very surprising... that makes it sound like it's doing some low-level parsing. Anyway, does it work if you instead add a percent line ending?

```latex
\tikzdeclarepattern{%
  name=lightcones,
```

Reply:
Yeah, it seems the linebreak is the problem, not all whitespace. I can push a commit changing this if you like!

Reply:
TeXLive 2018 patterns.meta code snippet:
`\def\pgfdeclarepattern#1{%
  \begingroup%
    \nullfont%
    \def\pgf@pat@options{#1}%
    \pgfkeys{/pgf/patterns/.cd, #1}%
    \pgf@declarepattern%
  \endgroup%
}
\def\pgf@declarepattern{%
    \pgfifpatternundefined{\pgf@pat@name}{%`

Latest patterns.meta code snippet:
`\def\pgfdeclarepattern#1{%
  \begingroup%
    \nullfont%
    \def\pgf@pat@options{,#1}%`
...
`    \pgf@declarepattern@meta%
  \endgroup%
}%
\def\pgf@declarepattern@meta{%`
...

In the new version, a comma has been added (fixing the linebreak issue, I **suspect**) and the helper command is renamed with @meta at the end to avoid overwriting the old pgf command. So both these were bugs in PGF/Tikz that have been fixed only (relatively) recently.